### PR TITLE
Fixes #3363 SslHandshakeCompletionHandler should use Netty promise

### LINKED ISSUE #3363: SslHandshakeCompletionHandler should use Netty promise
Since SslHandshakeCompletionHandler is a netty handler it should use netty promises instead of vertxHandler

### PR DESCRIPTION
**Pull request by zenios**

### CODE REVIEW

**Comment by vietj** [comment]
Can you rebase on latest master ? some files were changed - thanks

**Comment by zenios** [comment]
@vietj  done

File: src/main/java/io/vertx/core/net/impl/NetSocketImpl.java [hunk 1]
```diff
@@ -317,11 +315,19 @@ public NetSocket upgradeToSsl(Handler<AsyncResult<Void>> handler) {
...
+      p.addListener(future -> {
         if (handler != null) {
-          context.dispatch(ar.mapEmpty(), handler);
+          AsyncResult<Void> res;
+          if (future.isSuccess()) {
+            res = Future.succeededFuture();
+          } else {
+            res = Future.failedFuture(future.cause());
+          }
+          context.dispatch(null, u -> handler.handle(res));
```
Comment (vietj) [inline src/main/java/io/vertx/core/net/impl/NetSocketImpl.java:328]: here instead use `context.dispatch(res, handler)`

**Comment by zenios** [comment]
Done

**Pull request merged**